pudb/debugger.py

```diff
@@ -1492,38 +1492,25 @@ def add_cmdline_content(s, attr):
                     coming_from="above")
 
         def cmdline_tab_complete(w, size, key):
-            from rlcompleter import Completer
+            try:
+                from jedi import Interpreter
+            except ImportError:
+                add_cmdline_content("Tab completion requires jedi to be installed.",
+                                    "command line output")
 
             text = self.cmdline_edit.edit_text
             pos = self.cmdline_edit.edit_pos
 
             chopped_text = text[:pos]
             suffix = text[pos:]
 
-            # stolen from readline in the Python interactive shell
-            delimiters = " \t\n`~!@#$%^&*()-=+[{]}\\|;:\'\",<>/?"
-
-            complete_start_index = max(
-                    chopped_text.rfind(delim_i)
-                    for delim_i in delimiters)
-
-            if complete_start_index == -1:
-                prefix = ""
-            else:
-                prefix = chopped_text[:complete_start_index+1]
-                chopped_text = chopped_text[complete_start_index+1:]
-
-            state = 0
-            chopped_completions = []
-            completer = Completer(cmdline_get_namespace())
-            while True:
-                completion = completer.complete(chopped_text, state)
-
-                if not isinstance(completion, str):
-                    break
-
-                chopped_completions.append(completion)
-                state += 1
```

Comment:
Does Jedi not need all this logic?

Reply:
No jedi just returns a list of completions.